recipes/cgns/all/conanfile.py

```diff
@@ -85,6 +113,7 @@ def package(self):
         cmake.install()
 
         os.remove(os.path.join(self.package_folder, "include", "cgnsBuild.defs"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

Comment:
Hrm, so as the recipe stands, it will generate the target cgns::cgns, right?

Here is a repo with all the outputs from a shared+static build, could you please help to look at the targets generated?
https://github.com/paulharris/test-cgns-output

From what I can tell, it also generates:
* CGNS::cgns_static (for static builds)
* CGNS::cgns_shared (for shared builds)

`foreach(_expectedTarget CGNS::cgns_static)`

For shared, it also adds a define.  I've dealt with that in a different way in package_info
```
set_target_properties(CGNS::cgns_shared PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "USE_DLL"
```

So I should add components for these?
I'd like to stay with cgns::cgns for the "normal" target...
I'm not sure I should provide these other targets as they are the "old way".

Reply:
if you aim for flawless (=transparent) integration, you better provide these targets
for instance, if some library depends on `cgns` and uses `find_package(CGNS)` + `target_link_libraries(${PROJECT_NAME} PRIVATE CGNS::cgns_static)` it won't work out of the box with conan generator - you would need to patch `CMakeLists.txt`.
it's not mandatory to provide these components, but in general it's good practice. however, you can add it later, on demand, in another PR, it doesn't block an acceptance to CCI. (it's better to have recipe with no components over having no recipe at all).

Reply:
Ok, should be ok now?